global-script.lua

```diff
@@ -2896,20 +2931,120 @@ function upCastPosSizRot(oPos,size,rot,dist,multi,tags)
     end
     return hitObjects
 end
----- Block Square Section
+
+-- Updates the selected player color's player area.  Does nothing if they don't have one.
+-- Returns TRUE if an update occured, FALSE if no such player area existed.
+function updatePlayerArea(color)
+    local obj = playerBlocks[color]
+    if obj then
+        obj.call("setupPlayerArea")
+        return true
+    end
+    return false
+end
+-- Updates all player areas.
+function updateAllPlayerAreas()
+    for _,obj in pairs(playerBlocks) do
+        obj.call("setupPlayerArea")
+    end
+end
+
 function setupPlayerArea(params)
-    --Sets position/color for the button, spawns it
-    params.obj.createButton({
-        label="Energy Cost: 0", click_function="nullFunc",
-        position={0,2.24,-11.2}, rotation={0,180,0}, height=0, width=0,
-        font_color={1,1,1}, font_size=500
-    })
-    for _,bag in pairs(params.elementBags) do
-        bag.createButton({
-            label="0", click_function="nullFunc",
-            position={0,2.04,1.05}, rotation={0,0,0}, height=0, width=0,
-            font_color={1,1,1}, font_size=450
+    -- Figure out what color we're supposed to be, or if playerswapping is even allowed.
+    local obj = params.obj
+    local timer = obj.getVar("timer")  -- May be nil
+    local initialized = obj.getVar("initialized")
+    local color
+    for k, v in pairs(playerBlocks) do
+        if v.guid == obj.guid then
+            color = k
+            break
+        end
+    end
+    obj.setVar("playerColor", color)
+    local selected = selectedColors[color]
+
+    local playerReadyGuids = aidBoard.getTable("playerReadyGuids")
+    if not selected then
```

Comment:
what is this block of code trying to do?

Reply:
If you change from Red to Blue after having starting the game, the ready checker in the aidboard will show Red as "Not Ready".  Since Red no longer exists in selectedColors, it will always show that and never get updated (or cleared.)

This does a one-time-per-update removal of that text in the ready UI so you aren't stuck with it saying "Not Ready" for player colors that no longer exist.

Reply:
ah gotcha, this makes sense now